server/src/main/java/org/apache/cassandra/sidecar/tasks/PeriodicTaskExecutor.java

```diff
@@ -240,31 +263,22 @@ private void executeInternal(Promise<Void> promise, PeriodicTaskKey key, long ex
 
             case RESCHEDULE:
             default:
-                LOGGER.debug("Reschedule the task. task='{}' execCount={}", key, execCount);
-                reschedule(periodicTask);
-                promise.tryComplete();
+                LOGGER.debug("Rescheduling the task. task='{}' execCount={}", key, execCount);
+                promise.tryComplete(Result.RESCHEDULED);
```

Comment:
This is the fix. Do not reschedule, but just signal the decision and update the delay when scheduling a new run.